core/src/main/java/io/grpc/Context.java

```diff
@@ -759,7 +755,11 @@ private ExecutableListener(Executor executor, CancellationListener listener) {
     }
 
     private void deliver() {
-      executor.execute(this);
+      try {
+        executor.execute(this);
+      } catch (Throwable t) {
+        LOG.log(Level.INFO, "Exception notifying context listener", t);
```

Comment:
`Executor.execute()` does not throw `InterruptedException`, which as a checked exception.